ex16_praticepython.py

```diff
@@ -9,10 +9,15 @@
 import string
 import random
 
-def password(dlugosc,chars=string.ascii_letters+string.ascii_lowercase+string.punctuation+string.digits):
-    return "".join(random.choice(chars) for _ in range(dlugosc))
+def password(length,chars=string.ascii_letters+string.ascii_lowercase+string.punctuation+string.digits):
+#By command join we can join each string chars. 
+#In this situation we join without any letter or space, if we change to "*", we receive string1*string2*string3 
+#We can use "_" just as any other sign 
+    return "".join(random.choice(chars) for _ in range(length))

```

Comment:
- join() pracuje na tablicach
- jeśli będzie to tablica stringów np. ["ala", "ma", "kota"] to rzeczywiście zadziała tak jak piszesz
- w tym przypadku generujesz w locie tablicę z wykorzystaniem "list comprehension"
  - `for _ in range(length)` ustawia w karzdym kroku zmienną `_` na kolejną wartość z range
  - tak naprawdę nie wykorzystujesz wartości _ więc nie ma znaczenia że tak dziwnie się nazywa
  - w każdym kroku pętli w końcu ustawiasz wartość kolejnego elementu tablicy na lozowy znak z tablicy znaków `chars`

Prostszym rozwiązaniem byłoby: https://docs.python.org/3/library/random.html#random.choices